src/tabmat/ext/cat_split_helpers-tmpl.cpp

```diff
@@ -10,24 +10,29 @@ void _transpose_matvec_${dropfirst}(
     F* res,
     Int res_size
 ) {
-    #pragma omp parallel
+    int num_threads = omp_get_max_threads();
+    std::vector<F> all_res(num_threads * res_size, 0.0);
```

Comment:
Am I understanding correctly that we are using ~N times the memory compared to before (N being the number of threads)?
For transpose matvec this can be a problem because the result is as large as the input. Therefore, if the user has a very large categorical matrix as the main part of X, it will require N times more memory than X.

Reply:
I'm not the most familiar with OpenMP but was a new vector `restemp` not allocated per thread before (see https://github.com/Quantco/tabmat/blob/main/src/tabmat/ext/cat_split_helpers-tmpl.cpp#L15)?

Reply:
You are right. Then let's merge this. Thanks for the contribution!